src/frontend/js/widgets/Dashboard/components/DashboardOrderLoader/index.tsx

```diff
@@ -18,41 +20,56 @@ const messages = defineMessages({
     description: 'Banner displayed when the contract is not signed',
     id: 'components.DashboardOrderLoader.signatureNeeded',
   },
+  wrongLinkedProductError: {
+    defaultMessage: 'This page is not available for this order.',
+    description: "Banner displayed when order's linked product type is not handle.",
+    id: 'components.DashboardOrderLoader.wrongLinkedProductError',
+  },
 });
 
 export const DashboardOrderLoader = () => {
+  const [wrongLinkedProductError, setWrongLinkedProductError] = useState<string>();
   const params = useParams<{ orderId: string }>();
-  const order = useOmniscientOrder(params.orderId);
-  const course = order.item?.course as CourseLight;
-  const courseProduct = useCourseProduct(course?.code, { productId: order.item?.product });
-  const fetching = order.states.fetching || courseProduct.states.fetching;
-  const needsSignature = order.item?.contract && !order.item?.contract.signed_on;
+  const {
+    item: order,
+    states: { fetching: fetchingOrder, error: errorOrder },
+  } = useOmniscientOrder(params.orderId);
+  const {
+    states: { fetching: fetchingCourseProduct, error: errorCourseProduct },
+  } = useCourseProduct(order?.course?.code, { productId: order?.product });
+  const fetching = fetchingOrder || fetchingCourseProduct;
+  const needsSignature = order?.contract && !order?.contract.signed_on;
   const intl = useIntl();
 
+  const credentialOrder = order && isCredentialOrder(order) ? order : undefined;
+  useEffect(() => {
+    if (order && !credentialOrder) {
+      handle(new Error("Order's details page only accept order linked to CREDENTIAL product."));
+      setWrongLinkedProductError(intl.formatMessage(messages.wrongLinkedProductError));
+    }
+  }, [credentialOrder]);
+
+  const error = errorOrder || errorCourseProduct || wrongLinkedProductError;
```

Comment:
By convention we always put useEffect between variable declarations and return statement